pkg/secrets/secrets.go

```diff
@@ -23,6 +23,82 @@ var (
 	OverrideMountsFile = "/etc/containers/mounts.conf"
 )
 
+// SecretData info
+type SecretData struct {
+	Name string
+	Data []byte
+}
+
+// SaveTo saves secret data to given directory
+func (s SecretData) SaveTo(dir string) error {
+	path := filepath.Join(dir, s.Name)
+	if err := os.MkdirAll(filepath.Dir(path), 0700); err != nil && !os.IsExist(err) {
+		return err
+	}
+	return ioutil.WriteFile(path, s.Data, 0700)
```

Comment:
Should we be attempting to preserve permissions from the host?

Reply:
Should we? I don't think we ever did.